src/store/actions/UserProfile.ts

```diff
@@ -0,0 +1,22 @@
+import { ReduxAction } from "types/redux";
+
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+export enum UserProfileActionTypes {
+  SET_USER_PROFILE_DATA = "SET_USER_PROFILE_DATA",
+}
+
+export type SetUseProfileDataAction = ReduxAction<
+  UserProfileActionTypes.SET_USER_PROFILE_DATA,
+  { userProfile: WithId<User> | undefined }
+>;
+
+export const setUserProfileData = (
+  userProfile: WithId<User> | undefined
+): SetUseProfileDataAction => ({
+  type: UserProfileActionTypes.SET_USER_PROFILE_DATA,
+  payload: { userProfile },
+});
```

Comment:
Following the Redux styleguide, we shouldn't write our actions as 'setters' like this:

- https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters

(I realise this is how most of the existing code is written, but it's legacy and we want to change/remove that, so best not to add to the tech debt when we can help it)